### PR TITLE
Update ubi base image

### PR DESCRIPTION
**Pull request by dimakis**
# Issue link

# What changes have been made

updates base image as per [this conversation](https://redhat-internal.slack.com/archives/C05NXTEHLGY/p1697548522169089?thread_ts=1697036534.719189&cid=C05NXTEHLGY)

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.19.10-16 as builder
+FROM registry.redhat.io/ubi8/go-toolset:1.19 as builder
```
Comment (astefanutti) [inline Dockerfile:2]: Should `egistry.access.redhat.com` be kept?

Comment (astefanutti) [reply Dockerfile:2]: Thanks @jbusche, I agree with you.

Note these are pointing to the exact same image. It's only the access that changes.

Comment (astefanutti) [reply Dockerfile:2]: Actually CI is failing for that exact reason.

Comment (dimakis) [reply Dockerfile:2]: Thanks guys, I wan't aware. I'll update now

**Comment by jbusche** [comment]
This is a huge improvement over the rc.2 image, but it's not clear to me how much of it is due to the new go-toolset:1.19 image versus just doing a new rebuild to pickup the latest OS fixes from the ubi-minimal:8.8, but as you can see it's much better:

quay.io/project-codeflare/codeflare-operator:v1.0.0-rc.2
```
"vulnerabilityFailureSummary","C:1|H:5|M:34|L:6|T:46"
```
versus the one I just build from the Dimitri branch (once I fixed the io to com)
```
"vulnerabilityFailureSummary","C:1|H:1|M:6|L:0|T:8"
```

**Comment by jbusche** [comment]
I also stood it up on my server, and the new image is properly processing appwrappers:
```
All 20 appwrappers finished: 11:53:12
Total amount of time for 20 appwrappers is: 124 seconds
```
and
```
All 100 appwrappers finished: 12:09:10
Total amount of time for 100 appwrappers is: 594 seconds
```

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.19.10-16 as builder
+FROM registry.access.redhat.io/ubi8/go-toolset:1.19 as builder
```
Comment (astefanutti) [inline Dockerfile:2]: This should be `registry.access.redhat.com`?

**Pull request merged**